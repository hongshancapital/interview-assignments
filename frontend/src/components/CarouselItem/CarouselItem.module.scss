.carouselItem {
  min-height: 500px;
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  position: relative;
  box-sizing: border-box;
  white-space: pre-line;
  overflow: hidden;
}
.carouselItemTitle {
  font-size: 52px;
  z-index: 1;
  margin: 24vh 0 24px 0;
}
.carouselItemSubTitle {
  font-size: 28px;
  font-weight: normal;
  z-index: 1;
  margin: 0;
}
.carouselItemImage {
  position: absolute;
  bottom: 12vh;
}
.carouselItemIphone {
  background-color: #111;
  color: #fff;
  img {
    width: 660px;
  }
}
.carouselItemTablet {
  background: #fafafa;
  color: #111;
  img {
    width: 1200px;
  }
}
.carouselItemAirPod {
  background: #f1f1f3;
  color: #111;
  img {
    width: 1600px;
  }
}
