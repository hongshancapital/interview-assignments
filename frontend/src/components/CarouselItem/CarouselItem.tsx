import { FC } from 'react';
import styles from './CarouselItem.module.scss';

export type CarouselItemProps = {
  imgSrc: string;
  title: string;
  subTitle?: string;
  className?: string;
};

/**
 * @Component CarouselItem
 * @description Render a carousel item based on passed in props
 */
export const CarouselItem: FC<CarouselItemProps> = ({
  imgSrc,
  title,
  subTitle = '',
  className = '',
}) => {
  return (
    <div className={`${styles.carouselItem} ${className}`}>
      <h2 className={styles.carouselItemTitle}>{title}</h2>
      <h3 className={styles.carouselItemSubTitle}>{subTitle}</h3>
      <img className={styles.carouselItemImage} src={imgSrc} alt={title} />
    </div>
  );
};
