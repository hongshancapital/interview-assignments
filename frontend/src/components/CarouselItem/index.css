.my-carousel-item {
  display: block;
  width: 100%;
  height: 100%;
  white-space: normal;
}