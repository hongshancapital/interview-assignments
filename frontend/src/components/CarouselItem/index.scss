.prefix-carousel-item-wrap {
  position: relative;
  overflow: hidden;
}
.prefix-carousel-item {
  display: flex;
  flex-direction: column;
  position: absolute;
  top: 0;
  left: 50%;
  transform: translateX(-50%);
  margin-top: 180px;
  &-dark {
    color: #fff;
    background-color: #111;
  }
  &-light {
    color: #000;
    background-color: #fafafa;
  }
  &-gray {
    color: #000;
    background-color: #f1f1f3;
  }

  &-image {
    height: 50%;
    position: absolute;
    left: 50%;
    bottom: 80px;
    transform: translateX(-50%);
  }

  &-header {
    font-size: 56px;
    white-space: nowrap;
    font-weight: 600;
    margin: 0;
    &:last-of-type {
      margin-bottom: 20px;
    }
  }

  &-content {
    font-size: 28px;
    white-space: nowrap;
    font-weight: 400;
    margin: 0;
  }
}
