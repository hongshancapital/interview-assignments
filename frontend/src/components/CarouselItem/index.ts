export * from './CarouselItem';
export * from './constants';
