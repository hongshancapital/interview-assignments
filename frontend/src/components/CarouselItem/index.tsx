import classnames from 'classnames'
import React, { HTMLAttributes } from 'react'
import './index.scss'

export interface ICarouselItemProps extends HTMLAttributes<HTMLDivElement> {
  key: string
  header?: string[]
  content?: string[]
  imageSrc?: string
  theme?: 'dark' | 'light' | 'gray'
}
const classPrefix = 'prefix-carousel-item'
export default function CarouselItem(props: ICarouselItemProps) {
  const { key, header, content, imageSrc, theme, className, ...restProps } =
    props
  return (
    <div
      className={classnames(
        className,
        `${classPrefix}-wrap`,
        `${classPrefix}-${theme}`,
      )}
      {...restProps}
    >
      <img className={`${classPrefix}-image`} src={imageSrc} alt={key} />
      <div className={classnames(`${classPrefix}`)}>
        {!!header?.length &&
          header.map((title: string, index: number) => (
            <h2 className={`${classPrefix}-header`} key={index}>
              {title}
            </h2>
          ))}
        {!!content?.length &&
          content.map((text: string, index: number) => (
            <p className={`${classPrefix}-content`} key={index}>
              {text}
            </p>
          ))}
      </div>
    </div>
  )
}
