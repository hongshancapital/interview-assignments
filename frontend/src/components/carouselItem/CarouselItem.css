.carousel-item {
  min-width: 100%;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}