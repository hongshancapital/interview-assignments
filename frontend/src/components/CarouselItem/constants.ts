import { CarouselItemProps } from './CarouselItem';
import styles from './CarouselItem.module.scss';
import airPodImg from '../../assets/airpods.png';
import iphoneImg from '../../assets/iphone.png';
import tabletImg from '../../assets/tablet.png';

export const APP_CAROUSEL_ITEMS: CarouselItemProps[] = [
  {
    imgSrc: iphoneImg,
    title: 'xPhone',
    subTitle: `Lots to love.Less to spend.\nStarting at $399.`,
    className: styles.carouselItemIphone,
  },
  {
    imgSrc: tabletImg,
    title: 'Tablet',
    subTitle: `Just the right amount of everything.`,
    className: styles.carouselItemTablet,
  },
  {
    imgSrc: airPodImg,
    title: 'Buy a Tablet or xPhone for college.\nGet airPods.',
    className: styles.carouselItemAirPod,
  },
];
