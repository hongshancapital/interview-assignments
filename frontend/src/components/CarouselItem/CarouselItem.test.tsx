import { render } from '@testing-library/react';
import { CarouselItem } from './CarouselItem';

describe('CarouselItem', () => {
  const dummyProps = {
    imgSrc: 'test image',
    title: 'Title',
    subTitle: 'Sub Title',
    className: 'test-class',
  };

  it('renders correctly with given props', () => {
    const { container } = render(<CarouselItem {...dummyProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with default props', () => {
    const { container } = render(
      <CarouselItem imgSrc="test image" title="Test title" />
    );
    expect(container).toMatchSnapshot();
  });
});
