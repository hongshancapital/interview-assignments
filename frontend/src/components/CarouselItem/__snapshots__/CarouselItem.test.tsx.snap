// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem renders correctly with default props 1`] = `
<div>
  <div
    class="carouselItem "
  >
    <h2
      class="carouselItemTitle"
    >
      Test title
    </h2>
    <h3
      class="carouselItemSubTitle"
    />
    <img
      alt="Test title"
      class="carouselItemImage"
      src="test image"
    />
  </div>
</div>
`;

exports[`CarouselItem renders correctly with given props 1`] = `
<div>
  <div
    class="carouselItem test-class"
  >
    <h2
      class="carouselItemTitle"
    >
      Title
    </h2>
    <h3
      class="carouselItemSubTitle"
    >
      Sub Title
    </h3>
    <img
      alt="Title"
      class="carouselItemImage"
      src="test image"
    />
  </div>
</div>
`;
